src/androidTest/java/com/couchbase/lite/DatabaseTest.java

```diff
@@ -335,4 +338,73 @@ public void testAndroid2MLimit() throws Exception {
         assertNotNull(properties);
         assertEquals(content, properties.get("content"));
     }
+
+    // Database_Tests.m : test18_Attachments
+    public void testAttachments() throws Exception {
+        Map<String, Object> properties = new HashMap<String, Object>();
+        properties.put("testName", "testAttachments");
+        properties.put("count", 1);
+
+        final Document doc = createDocumentWithProperties(database, properties);
+        SavedRevision rev = doc.getCurrentRevision();
+        assertEquals(0, rev.getAttachments().size());
+        assertEquals(0, rev.getAttachmentNames().size());
+        assertNull(rev.getAttachment("index.html"));
+
+        String content = "This is a test attachments!";
+        ByteArrayInputStream body = new ByteArrayInputStream(content.getBytes());
+        UnsavedRevision rev2 = doc.createRevision();
+        rev2.setAttachment("index.html", "text/plain; charset=utf-8", body);
+
+        assertEquals(1, rev2.getAttachments().size());
+        assertEquals(1, rev2.getAttachmentNames().size());
+        assertEquals("index.html", rev2.getAttachmentNames().get(0));
+        Attachment attach = rev2.getAttachment("index.html");
+        assertNotNull(attach);
+        assertNull(attach.getRevision()); // No revision set
+        assertNull(attach.getDocument()); // No revision set
```

Comment:
It's technically correct and inline with the iOS because only revision is set to the document and the document will be get from the set revision.